src/bootstrap/compile.rs

```diff
@@ -1246,7 +1246,7 @@ impl Step for Assemble {
 
     fn make_run(run: RunConfig<'_>) {
         run.builder.ensure(Assemble {
-            target_compiler: run.builder.compiler(run.builder.top_stage + 1, run.target),
+            target_compiler: run.builder.compiler(run.builder.top_stage, run.target),
```

Comment:
Presumably with something like --stage 0, this now is a no-op rather than doing something?

Reply:
As I mentioned earlier, `bootstrap::compile::Assemble::make_run` only gets called when a target dir is specified something like `x build compiler/rustc`.

When I execute `x build compiler/rustc --stage 0`, I get the exactly same `stage0` output compare to master branch, but missing `stage1`(which includes a bunch of empty directories), `ci-rustc`, `ci-rustc-sysroot` directories.